app/asset-manifest.js

```diff
@@ -1,13 +1,22 @@
 import environment from './config/environment';
 
-const metaName = environment.modulePrefix + '/asset-manifest';
+const modulePrefix = environment.modulePrefix;
+const metaName = `${modulePrefix}/asset-manifest`;
+const nodeName = `${modulePrefix}/node-asset-manifest`;
+
 let config = {};
 
 try {
-  const rawConfig = document.querySelector('meta[name="' + metaName + '"]').getAttribute('content');
-  config = JSON.parse(unescape(rawConfig));
+  // If we have a Node version of the asset manifest, use that for FastBoot and
+  // similar environments.
+  if (require.has(nodeName)) {
```

Comment:
I'm unsure I like that approach, since that means we'll essentially have 2 `try-catch` blocks, since detecting whether or not the meta approach will work is kind of spotty.